h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -146,6 +146,10 @@ public long sizeAsLong() {
         if (undoLogsTotalSize == 0) {
             return size;
         }
+        // if we are in READ_UNCOMMITTED, no need to check the undologs
+        if (transaction.isReadUncommitted()) {
```

Comment:
With `if (!transaction.isReadCommitted())` new method can be removed.

Reply:
I fail to understand why in READ_UNCOMMITTED, **no need to check the undologs**. If another transaction (or even yours) just removed a bunch of records, but not committed yet, will you still count them? Because until commit they are still contributing into total map size with their tombstones. What is COUNT(*) in READ_UNCOMMITTED anyway? At least, you should account for your own deletions.